src/bson_mapper/test/utility.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "catch.hpp"
+
+#include <bson_mapper/utility.hpp>
+
+#include <iostream>
+
+using namespace bson_mapper;
+using namespace bsoncxx;
+
+class Foo {
+   public:
+    int a, b, c;
+
+    bool operator==(const Foo &rhs) {
+        return (a == rhs.a) && (b == rhs.b) && (c == rhs.c);
+    }
+
+    // void serialize(Archive& ar) {
+    //     ar(CEREAL_NVP(a), CEREAL_NVP(b), CEREAL_NVP(c), CEREAL_NVP(f));
+    // }
+};
+
+// set up test BSON documents and objects
+std::string json_str = "{\"a\": 1, \"b\":4, \"c\": 9}";
```

Comment:
You can make this look cleaner by using [raw string literals](http://en.cppreference.com/w/cpp/language/string_literal):
```
std::string json_str = R"({"a": 1, "b":4, "c": 9})"
```